api/files.py

```diff
@@ -34,9 +34,20 @@ def getHashingFieldStorage(upload_dir, hash_alg):
     class HashingFieldStorage(cgi.FieldStorage):
         bufsize = 2**20
         def make_file(self, binary=None):
+            if not self.filename:
+                return self.file
             self.open_file = HashingFile(os.path.join(upload_dir, os.path.basename(self.filename)), hash_alg)
             return self.open_file
 
+        # override private method __write of superclass FieldStorage
+        # _FieldStorage__file is the private variable __file of the same class
+        def _FieldStorage__write(self, line):
+            if self._FieldStorage__file is not None:
+                self.file = self.make_file('')
+                self.file.write(self._FieldStorage__file.getvalue())
```

Comment:
Since `.getvalue()` returns the entire contents of the file, are you certain this branch of logic will only trigger on inconsequentially small files? How could we validate that?